### PR TITLE
#162685680 Comment Should Track Edit History

### PR DESCRIPTION
**Pull request by moseskamira**
#### What Does This PR Do?
The PR delivers a feature that enables a user to track edit history of a given comment
#### Description of the task to be completed
The task shall be used whenever a user wants to see edit history to a specific comment.
#### Link To Pivotal Tracker Story
#162685680

#### How Should This Be Manually Tested
Before testing run these commands in cmd
git clone https://github.com/andela/Ah-backend-guardians.git
cd Ah-backend-guardians
git checkout ft-edit-history-162685680
virtualenv env
. env/bin/activate
pip install -r requirements.txt
##### Comment on the article by POST
Url: http://127.0.0.1:8000/api/slug/comments/1/
```
{
  "body": "It is a very good idea"
}
```
##### Update the comment by a PUT
URL: http://127.0.0.1:8000/api/slug/comments/1/
```
{
  "body": "It is a very good idea"
}
```
#43 #### Fetch edit history of comment by a GET
URL: http://127.0.0.1:8000/api/slug/comments/1/history/
```
{
    "comment": [
        {
            "id": 1,
            "body": "What is going on here",
            "created_at": "2019-02-01T12:03:54.055803Z",
            "user": 1
        },
        {
            "id": 2,
            "body": "What is going on here, we go so",
            "created_at": "2019-02-01T12:04:02.897834Z",
            "user": 1
        },
        {
            "id": 3,
            "body": "What is going on here, we go so and now and now",
            "created_at": "2019-02-01T12:18:27.546739Z",
            "user": 1
        },
        {
            "id": 4,
            "body": "Tom is here",
            "created_at": "2019-02-01T12:19:12.696280Z",
            "user": 1
        },
        {
            "id": 5,
            "body": "Tom is here with Moses",
            "created_at": "2019-02-01T12:26:12.443134Z",
            "user": 1
        },
        {
            "id": 6,
            "body": "Tom is here with Moses again",
            "created_at": "2019-02-01T12:26:28.692308Z",
            "user": 1
        }
    ]
}
```

### CODE REVIEW

File: authors/apps/comments/models.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from django.db import models
 from authors.apps.authentication.models import User
 from authors.apps.articles.models import Article
+from django.contrib.contenttypes.fields import GenericRelation
```
Comment (Opio-Emmanuel-Omona) [inline authors/apps/comments/models.py]: @moseskamira You imported GenericRelation but never used it

File: authors/apps/comments/renderers.py [hunk 2]
```diff
@@ -15,3 +16,15 @@ def render(self, data, media_type=None, renderer_context=None):
...
             "likes count": len(data)
         })
+
+
+class CommentJSONRenderer(JSONRenderer):
+    charset = 'utf-8'
+
+    def render(self, data, media_type=None, renderer_context=None):
+
+        if type(data) != ReturnList:
```
Comment (Opio-Emmanuel-Omona) [inline authors/apps/comments/renderers.py]: @moseskamira I would like to know what happens when the type(data) is ReturnList

File: authors/apps/comments/serializers.py [hunk 3]
```diff
@@ -16,3 +16,25 @@ class LikeCommentSerializer(serializers.ModelSerializer):
...
+
+
+class CommentHistorySerializer(serializers.ModelSerializer):
+    """
+    Class Handling History of Comment Edited
+    """
+
+    class Meta:
+        model = EditHistory
+        fields = ('id', 'body', 'created_at', 'user')
```
Comment (Opio-Emmanuel-Omona) [inline authors/apps/comments/serializers.py:28]: @moseskamira It appears that there are two classes with the same name. i.e.
```class CommentHistorySerializer(serializers.ModelSerializer):```

File: authors/apps/comments/tests/test_edit_history.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+
+        response2 = self.client.get(self.url2)
+        self.assertEqual(response2.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            response2.data['message'],
+            "No Edit History For To This Comment")
+
+    def test_comment_does_not_exist(self):
+        """TTesting that comment does not exist"""
```
Comment (Opio-Emmanuel-Omona) [inline authors/apps/comments/tests/test_edit_history.py]: @moseskamira 
```suggestion
        """Testing that comment does not exist"""
```

File: authors/apps/comments/serializers.py (hunk 3 above)
Comment (moseskamira) [reply authors/apps/comments/serializers.py:28]: Worked upon it

File: authors/apps/comments/tests/test_edit_history.py (hunk 4 above)
Comment (moseskamira) [reply authors/apps/comments/tests/test_edit_history.py]: Solved

**Review by fahadmak — APPROVED** [review]
LGTM

**Review by Opio-Emmanuel-Omona — APPROVED** [review]
LGTM

**Pull request merged**